gsy_e_sdk/setups/Asset_API_scripts/canary_network_live_data.py

```diff
@@ -0,0 +1,143 @@
+# flake8: noqa
+"""
+Template file to send live data to a Canary Network using Rest.
+"""
+import os
+from time import sleep
+from pendulum import from_format
+from gsy_framework.constants_limits import DATE_TIME_FORMAT
+from gsy_e_sdk.aggregator import Aggregator
+from gsy_e_sdk.clients.rest_asset_client import RestAssetClient
+from gsy_e_sdk.utils import get_area_uuid_from_area_name_and_collaboration_id
+
+current_dir = os.path.dirname(__file__)
+ORACLE_NAME = "oracle"
+
+LOAD_NAMES = ["Load 1", "Load 2"]
+PV_NAMES = ["PV 1", "PV 2"]
+STORAGE_NAMES = []
+AUTOMATIC = False
+
+
+class Oracle(Aggregator):
+    """Class that defines the behaviour of an "oracle" aggregator."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+
+    def on_market_cycle(self, market_info):
+        """Place a bid or an offer whenever a new market is created."""
+        if self.is_finished is True:
+            return
+        self.send_forecasts(market_info)
+
+    def on_tick(self, tick_info):
+        pass
+
+    def send_forecasts(self, market_info):
+        """Send forecasts of the next market slot to the exchange."""
+        forecast_market_slot = (
+            from_format(market_info["market_slot"], DATE_TIME_FORMAT)
+            .add(minutes=15)
+            .format(DATE_TIME_FORMAT)
+        )
+        # pylint: disable=unused-variable
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            asset_info = area_dict.get("asset_info")
+            if not asset_info:
+                continue
+
+            # Consumption assets
+            if "energy_requirement_kWh" in asset_info:
+                asset_name = area_dict["area_name"]
+                globals()[f"{asset_name}"].set_energy_forecast(
```

Comment:
Interesting! I am not sure whether this is best practice, but I see the need for it, so keep it as is.
(I ignore the case when `f"{asset_name}"` is not in the global variabel buffer, you probably want to raise an exception then, right?)

Reply:
Could you maybe elaborate on how you would introduce this exception? Thanks a lot!

Reply:
Sorry, I formulated not clear. Let me try again:
If you have not added the `f"{asset_name}"` key value pair to the `globals()`, an exception will be raised. This is wanted behaviour, right?